### PR TITLE
Added workflow

### PR DESCRIPTION
**Pull request by soumyaDghosh**
Also adds support for dev releases. Adds `-dev` after the version if it's a dev version. Also, doesn't creates a call-for-testing issue for dev versions and releases dev versions to `edge` channel instead of candidate.

### CODE REVIEW

**Comment by soumyaDghosh** [comment]
Branch renaming is needed. I was also thinking to move to `core22`, but regarding classic snaps, I am currently unsure, if they'll work everywhere

**Comment by popey** [comment]
Please do *not* move to core22. You will break the application.

**Comment by soumyaDghosh** [comment]
No I didn't it's still on core18

**Comment by popey** [comment]
I know, you said you're thinking of it. I'm saying "don't".
Also, please do not merge this until the snap is tested on Ubuntu 20.04 and 22.04, not just 23.10. 
We have already had the sublime-text snap break, and we should be cautious about changing the entire build process.

I'm also uneasy that you've changed the build process in among a process to add workflow. They are separate things, and should be separate PRs.

**Comment by soumyaDghosh** [comment]
> I'm also uneasy that you've changed the build process in among a process to add workflow. They are separate things, and should be separate PRs.

I made the `source` of this snap the same as `sublime-text`

https://github.com/snapcrafters/sublime-text/blob/candidate/snap/snapcraft.yaml#L37

Previously `tar` file was used, and just changed it to the deb file. And also added a support for the `dev` builds. There is nothing I changed that is probably dangerous/breaking.

**Comment by popey** [comment]
> They are separate things, and should be separate PRs.

**Comment by jnsgruk** [comment]
In the spirit of keeping this lean and easy to review; can we also drop the change that introduces `dev` versions? Let's get this bought up to date with the latest workflows that release to candidate first, then we discuss options for `dev` versions later?

File: .github/workflows/release-to-candidate.yml
Comment (jnsgruk) [inline .github/workflows/release-to-candidate.yml]: Per my latest comment on the PR, this section feels problematic in a workflow that only triggers on pushes to the `candidate` branch. I think we should split this out and discuss separately.

Comment (soumyaDghosh) [reply .github/workflows/release-to-candidate.yml]: Okay.. I am moving it too.

File: .github/workflows/sync-upstream.yml
Comment (jnsgruk) [inline .github/workflows/sync-upstream.yml]: Feels a bit heavyweight to introduce python here (with a non-std lib requirement on `requests`, no less). Can we rework this to just use shell?

Comment (jnsgruk) [reply .github/workflows/sync-upstream.yml]: For example:

```
$ curl -sL https://www.sublimemerge.com/download | grep -Po '<p class="latest"><i>Version:</i> Build \K[0-9]+'
2091

$ curl -sL https://www.sublimemerge.com/dev | grep -Po '<p class="latest"><i>Version:</i> Build \K[0-9]+'
2092
```

Comment (soumyaDghosh) [reply .github/workflows/sync-upstream.yml:25]: But, we'll not able to compare them, like for eg. the `dev` version for `sublime-text` is `4168`  but stable one is `4169`, this one has the `dev` channel a bigger number for now, but how much can we rely on that. If we don't plan at all to support `dev` versions, (There was an issue about that in the sublime-text repo to add dev versions), then yes, definitely yes to shell.

Comment (soumyaDghosh) [reply .github/workflows/sync-upstream.yml:25]: http://mywiki.wooledge.org/ArithmeticExpression

this might come handy though.

Comment (soumyaDghosh) [reply .github/workflows/sync-upstream.yml]: Done!

File: .github/workflows/sync-upstream.yml
Comment (jnsgruk) [inline .github/workflows/sync-upstream.yml]: Would be good to tidy up the whitespace here and just end the file with a single newline

File: .github/workflows/sync-upstream.yml
Comment (jnsgruk) [inline .github/workflows/sync-upstream.yml]:
```suggestion
            stable_version="$(curl -sL https://www.sublimemerge.com/download | grep -Po '<p class="latest"><i>Version:</i> Build \K[0-9]+')"
```

File: .github/workflows/sync-upstream.yml
Comment (jnsgruk) [inline .github/workflows/sync-upstream.yml]: I'd recommend against using `yq` here, as it will reformat a lot of the yaml file.

```suggestion
            sed -i 's/^\(version: \).*$/\1'"$stable_version"'/' snap/snapcraft.yaml
```

File: snap/snapcraft.yaml
Comment (jnsgruk) [inline snap/snapcraft.yaml]: Can you add a newline on the end of this file please?

**Review by jnsgruk — CHANGES_REQUESTED** [review]
Some minor changes, but looking much more manageable :)

File: snap/snapcraft.yaml
Comment (soumyaDghosh) [reply snap/snapcraft.yaml]: Sure.. doing all the formatting

File: .github/workflows/sync-upstream.yml
Comment (soumyaDghosh) [reply .github/workflows/sync-upstream.yml]: sure

File: .github/workflows/sync-upstream.yml
Comment (soumyaDghosh) [reply .github/workflows/sync-upstream.yml]: I saw that thing happen in the ffmpeg workflow today, so, should I replace all the yqs (in other repos) with sed?

File: .github/workflows/sync-upstream.yml
Comment (jnsgruk) [inline .github/workflows/sync-upstream.yml]:
```suggestion

```

**Review by jnsgruk — DISMISSED** [review]
LGTM, thanks

**Comment by jnsgruk** [comment]
Ah, one last thing before merging - please remove `.github/workflows/test-snap-can-build.yml` - we don't need two workflows to build on PRs :)

File: .github/workflows/sync-upstream.yml
Comment (soumyaDghosh) [reply .github/workflows/sync-upstream.yml]: ?

**Review by jnsgruk — APPROVED** [review]

**Comment by soumyaDghosh** [comment]
@jnsgruk Just wait a minute before merging, I am testing the `source-change`. After that merge both together, as this workflow can't work in this current scenario

**Comment by soumyaDghosh** [comment]
Ahh.. merged it.. okay coming up with the second PR

**Pull request merged**
